### PR TITLE
The Witness: Migrate joke hints to the client

### PR DESCRIPTION
**Pull request by hatkirby**
## What is this fixing or adding?
Joke hints are being moved into the client, so they no longer need to be stored here. This change also nulls out joke hints sent to the client, which is needed for it to know to generate a joke hint locally.

Client PR: NewSoupVi/The-Witness-Randomizer-for-Archipelago#28

## How was this tested?
Test generations with and without audio log hints. Game was closed and reopened to ensure consistency.

### CODE REVIEW

**Comment by NewSoupVi** [comment]
Waiting on client PR: https://github.com/NewSoupVi/The-Witness-Randomizer-for-Archipelago/pull/28

**Review by NewSoupVi — CHANGES_REQUESTED** [review]
Can't request changes to that line but

```
        credits_hint = (
            "This Randomizer is brought to you by\n"
            "NewSoupVi, Jarno, blastron,\n"
            "jbzdarkid, sigma144, IHNN, oddGarrett, Exempt-Medic, hatkirby.", -1, -1
        )
```

**Comment by hatkirby** [comment]
That's correct.

**Review by NewSoupVi — APPROVED** [review]
https://github.com/NewSoupVi/The-Witness-Randomizer-for-Archipelago/pull/28 is merged. All of my concerns have been addressed.

**Review by NewSoupVi — APPROVED** [review]

File: worlds/witness/__init__.py [hunk 1]
```diff
@@ -335,15 +328,10 @@ def fill_slot_data(self) -> dict:
...
-            audio_log = audio_logs.pop()
-            self.log_ids_to_hints[int(audio_log, 16)] = credits_hint
-
-        joke_hints = generate_joke_hints(self, len(audio_logs))
-
+        # Client will generate joke hints for these.
         while audio_logs:
             audio_log = audio_logs.pop()
-            self.log_ids_to_hints[int(audio_log, 16)] = joke_hints.pop()
+            self.log_ids_to_hints[int(audio_log, 16)] = ("", -1, -1)
```
[the remark this answers is not in the document]
Comment (NewSoupVi) [reply worlds/witness/__init__.py:334]: Like, have it use a for loop or even `self.log_ids_to_hints.update({int(audio_log, 16): ("", -1, -1) for audio_log in audio_logs})`

**Comment by NewSoupVi** [comment]
Moving this to core review because it adds 3 lines

**Review by Exempt-Medic — APPROVED** [review]
My only comment got addressed in the past hour. The changes all LGTM and it still generates with and without joke hints.

**Comment by NewSoupVi** [comment]
This PR will likely conflict with https://github.com/ArchipelagoMW/Archipelago/pull/3007, which I absolutely need to be merged before anything else.

**Review by Berserker66 — APPROVED** [review]

**Pull request merged**